prysmaticlabs/prysm
# Implement delete gaslimit

**What type of PR is this?**
> Feature #11155 - delete gas limit

**What does this PR do? Why is it needed?**

**Which issues(s) does this PR fix?**

#11155 subtask -delete gas limit

> Tested:


1. set gaslimit

```
curl --header "Authorization: Bearer ${bearer}" \
  --header "Content-Type: application/json"  \
  --request POST \
  --data '{ "gas_limit": "30000002" }' \
  --write-out "%{http_code}\n" \
  127.0.0.1:7500/eth/v1/validator/${pubkey}/gas_limit
```
2. check "set gaslimit" is done correctly
```
curl --header "Authorization: Bearer ${bearer}" \
  --request GET \
  127.0.0.1:7500/eth/v1/validator/${pubkey}/gas_limit
```

3. delete gaslimit
```
curl --header "Authorization: Bearer ${bearer}" \
  --header "Content-Type: application/json"  \
  --request DELETE \
  --write-out "%{http_code}\n" \
  127.0.0.1:7500/eth/v1/validator/${pubkey}/gas_limit
```

4. make sure gas limit is deleted
```
curl --header "Authorization: Bearer ${bearer}" \
  --request GET \
  127.0.0.1:7500/eth/v1/validator/${pubkey}/gas_limit
```

Note: "delete gas limit" action here just sets gaslimit to 0 (uint64t(-1) is another option), a maybe better alternative is to change gaslimit from value type to pointer type, so we can use "nil" to clearly mark a non-existent gaslimit field.
